terraform/services/api-rds/main.tf

```diff
@@ -81,16 +81,17 @@ locals {
 # Create database security group
 resource "aws_security_group" "sg_database" {
   name = local.sg_name
-  description = var.app == "ab2d" ? "${local.db_name} database security group" : (
-  var.app == "dpc" ? "Security group for DPC DB" : "App ELB security group")
+  description = var.legacy ? (var.app == "ab2d" ? "${local.db_name} database security group" : (
+    var.app == "dpc" ? "Security group for DPC DB" : "App ELB security group")
+  ) : "For ${local.db_name} api database"
 
-  vpc_id = module.vpc.id
+  vpc_id = var.legacy ? module.vpc[0].id : module.platform[0].vpc_id
 
-  tags = merge(
+  tags = var.legacy ? merge(
     data.aws_default_tags.data_tags.tags,
     { "Name" = local.sg_name },
     var.app == "dpc" ? local.dpc_specific_tags : {}
-  )
+  ) : {}
```

Comment:
I think the tag Name is traditional for security groups. I think you're going to want those everywhere.

Also, you probably ought to drop `data.aws_default_tags.data_tags` all together–I don't think there's a single legitimate use for them in this module.